packages/fields/src/fields/featured-image/featured-image-edit.tsx

```diff
@@ -0,0 +1,122 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Button,
+	__experimentalGrid as Grid,
+	__experimentalText as Text,
+} from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { useCallback, useRef } from '@wordpress/element';
+// @ts-ignore
+import { MediaUpload } from '@wordpress/block-editor';
+import { lineSolid } from '@wordpress/icons';
+import { store as coreStore } from '@wordpress/core-data';
+import type { DataFormControlProps } from '@wordpress/dataviews';
+/**
+ * Internal dependencies
+ */
+import type { BasePost } from '../../types';
+import { __ } from '@wordpress/i18n';
+
+export const FeaturedImageEdit = ( {
+	data,
+	field,
+	onChange,
+}: DataFormControlProps< BasePost > ) => {
+	const { id } = field;
+
+	const value = field.getValue( { item: data } );
+
+	const media = useSelect(
+		( select ) => {
+			const { getEntityRecord } = select( coreStore );
+			return getEntityRecord( 'root', 'media', value );
+		},
+		[ value ]
+	);
+
+	const onChangeControl = useCallback(
+		( newValue: number ) =>
+			onChange( {
+				[ id ]: newValue,
+			} ),
+		[ id, onChange ]
+	);
+
+	const url = media?.source_url;
+	const title = media?.title?.rendered;
+	const ref = useRef( null );
+
+	return (
+		<fieldset className="fields-controls__featured-image">
```

Comment:
What do you think about this class name? I asked a similiar question in this PR: https://github.com/WordPress/gutenberg/pull/65196#discussion_r1764760161